modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItReplicasTest.java

```diff
@@ -101,7 +139,65 @@ void testLearnerReplicaCreatedAfterStartingNewNode() {
         });
     }
 
-    private static Set<Assignment> getStableAssignments(Ignite node, String tableName) {
+    @Test
+    void testLearnerReplicaReadOnlyRequestCanReadData() {
+        executeSql("CREATE ZONE TEST_ZONE (PARTITIONS 1, REPLICAS ALL, QUORUM SIZE 2) STORAGE PROFILES ['default']");
+        executeSql("CREATE TABLE TEST (id INT PRIMARY KEY, name INT) ZONE TEST_ZONE");
+        executeSql("INSERT INTO TEST VALUES (0, 0)");
+
+        cluster.startNode((int) cluster.runningNodes().count()); // starting a new node adds a learner
+        AtomicReference<IgniteImpl> learnerRef = new AtomicReference<>();
+
+        await().atMost(Duration.ofSeconds(30)).untilAsserted(() -> {
+            Set<Assignment> stableAssignments = stablePartitionAssignments(cluster.node(0), "TEST");
+            String learner = stableAssignments.stream().filter(a -> !a.isPeer()).map(Assignment::consistentId).findFirst().orElse(null);
+            assertTrue(learner != null && !learner.isBlank(), "learners found after staring new node");
+            learnerRef.set(findNode(node -> node.name().equals(learner)));
+        });
+
+        IgniteImpl learner = learnerRef.get();
+        ReplicaListener replicaListener = of(learnerRef.get()).map(toReplicaListener("TEST_ZONE", "TEST", 0)).orElseThrow();
+        ClusterNode learnerClusterNode = learner.clusterService().topologyService().localMember();
+        SchemaDescriptor schema = unwrapTableViewInternal(learner.tables().table("TEST")).schemaView().lastKnownSchema();
+        KvMarshaller<Integer, Integer> marshaller = new ReflectionMarshallerFactory().create(schema, Integer.class, Integer.class);
+        Row pk = marshaller.marshal(0);
```

Comment:
I would recommend to make a separate serialization method (or method returning marshaller)